spring-boot-starter-camunda-sdk/src/main/java/io/camunda/zeebe/spring/client/configuration/ZeebeClientConfigurationSpringImpl.java

```diff
@@ -175,43 +277,86 @@ public ScheduledExecutorService jobWorkerExecutor() {
+        properties::ownsJobWorkerExecutor,
+        DEFAULT.ownsJobWorkerExecutor(),
+        configCache);
   }
 
   @Override
   public boolean getDefaultJobWorkerStreamEnabled() {
-    return properties.getDefaultJobWorkerStreamEnabled();
+    return getOrLegacyOrDefault(
+        "DefaultJobWorkerStreamEnabled",
+        () -> camundaClientProperties.getZeebe().getDefaults().getStreamEnabled(),
+        properties::getDefaultJobWorkerStreamEnabled,
+        DEFAULT.getDefaultJobWorkerStreamEnabled(),
+        configCache);
   }
 
   @Override
   public boolean useDefaultRetryPolicy() {
-    return properties.useDefaultRetryPolicy();
+    return false;
   }
 
   @Override
   public boolean preferRestOverGrpc() {
-    return false;
+    return camundaClientProperties.getZeebe().isPreferRestOverGrpc();
   }
 
-  private CredentialsProvider initCredentialsProvider() {
+  private CredentialsProvider identityCredentialsProvider() {
+    if (!(authentication instanceof DefaultNoopAuthentication)) {
+      return new IdentityCredentialsProvider(authentication);
+    }
+    return null;
+  }
 
-    if (commonConfigurationProperties.getEnabled()
-        && !(authentication instanceof DefaultNoopAuthentication)) {
-      return new CredentialsProvider() {
-        @Override
-        public void applyCredentials(final CredentialsApplier applier) {
-          final Map.Entry<String, String> authHeader = authentication.getTokenHeader(Product.ZEEBE);
-          applier.put(authHeader.getKey(), authHeader.getValue());
-        }
+  private String composeGatewayAddress() {
+    // check if port is set
+    if (camundaClientProperties.getZeebe().getGatewayUrl().getPort() != -1) {
+      final String gatewayAddress =
+          camundaClientProperties.getZeebe().getGatewayUrl().getHost()
+              + ":"
+              + camundaClientProperties.getZeebe().getGatewayUrl().getPort();
+      LOG.debug("Gateway port is set, address will be '{}'", gatewayAddress);
+      return gatewayAddress;
+    }
+    // check if default port can be applied
+    if (camundaClientProperties.getZeebe().getGatewayUrl().getDefaultPort() != -1) {
+      final String gatewayAddress =
+          camundaClientProperties.getZeebe().getGatewayUrl().getHost()
+              + ":"
+              + camundaClientProperties.getZeebe().getGatewayUrl().getDefaultPort();
+      LOG.debug("Gateway port has default, address will be '{}'", gatewayAddress);
+      return gatewayAddress;
+    }
+    LOG.debug(
+        "Gateway cannot be determined, address will be '{}'",
+        camundaClientProperties.getZeebe().getGatewayUrl().getHost());
+    // do not use any port
+    return camundaClientProperties.getZeebe().getGatewayUrl().getHost();
+  }
 
-        @Override
-        public boolean shouldRetryRequest(final StatusCode statusCode) {
-          return statusCode.code() == Code.DEADLINE_EXCEEDED.value();
-        }
-      };
+  private boolean composePlaintext() {
```

Comment:
🔧 

```suggestion
  private boolean composePlaintext() {
    String protocol = camundaClientProperties.getZeebe().getGatewayUrl().getProtocol();
    switch (protocol) {
        case "http":
            return true;
        case "https":
            return false;
        default:
            throw new IllegalStateException(
                    String.format("Unrecognized zeebe protocol '%s'", protocol));
    }
}
```

Reply:
Introduced `switch` in my latest commit, thank you 👍